decidim-core/app/controllers/concerns/decidim/needs_permission.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require "active_support/concern"
+
+module Decidim
+  # Common logic to work with the permissions system
+  module NeedsPermission
+    extend ActiveSupport::Concern
+
+    included do
+      helper_method :allowed_to?
+
+      class ::Decidim::MissingPermission < StandardError
+      end
+
+      rescue_from Decidim::MissingPermission, with: :user_not_authorized
+
+      def check_permission_to(action, subject, extra_context = {})
+        raise Decidim::MissingPermission unless allowed_to?(action, subject, extra_context)
+      end
+
+      def allowed_to?(action, subject, extra_context = {})
+        permission_class.new(
+          current_user,
+          { scope: permission_scope.to_s, action: action.to_s, subject: subject.to_s },
+          ability_context.merge(extra_context)
```

Comment:
Should we rename `ability_context` to `permissions_context`?

Reply:
Or deprecate it.

Reply:
I think this is being done to be able to live with both permissions/abilities sub-systems while the rest of the app implements it.

Reply:
The `ability_context` method is provided by the `NeedsAuthorization` concern, and is overwritten at different controllers, like the `Decidim::Comnponents::BaseController` and others. I wanted to reuse the logic provided by that method, but didn't want to change everything yet, that's why I'm using the `ability_context` method directly.

Maybe, by now, I could make the `NeedsPermission` concern alias this method to `permissions_context` and do a full rename when all abilities are removed.

Does this work for you?

Reply:
👍 on the alias!